src/op_analytics/datasources/chainsrpc/erc20metadata.py

```diff
@@ -0,0 +1,188 @@
+from dataclasses import dataclass
+from functools import cache
+
+import polars as pl
+import requests
+
+from op_analytics.datapipeline.chains.load import load_chain_metadata
+from op_analytics.coreutils.clickhouse.oplabs import run_query_oplabs
+from op_analytics.coreutils.threads import run_concurrently
+from op_analytics.coreutils.logger import structlog
+
+
+log = structlog.get_logger()
+
+
+@dataclass(frozen=True)
+class TokenContract:
+    chain: str
+    chain_id: int
+    contract_address: str
+
+
+def execute_pull():
+    # Find the list of ERC20 tokens that are not yet in our metadata table.
+    PERCENTILE = 0.90
+
+    query = f"""
+        SELECT
+            derived.chain,
+            derived.chain_id,
+            derived.contract_address,
+            derived.num_transfers
+        FROM (
+            SELECT
+                t.chain,
+                t.chain_id,
+                CAST(t.contract_address AS String) AS contract_address,
+                t.num_transfers,
+                quantile({PERCENTILE})(t.num_transfers) OVER (PARTITION BY t.chain, t.chain_id) AS percentile_num_transfers
```

Comment:
if this ever hits performance issues, I'm sure we can do something like "where transfers > 10" as well. But not necessary if it works fine.